### PR TITLE
Snap 1047

### PR DESCRIPTION
**Pull request by rishitesh**
## Changes proposed in this pull request
Changed cached insert logic to be extendible . 
Updated PartitionRegionStats to reflect true row count.
This stats can be used by vsd , mbeans to reflect true row count.
## Patch testing

precheckin

## ReleaseNotes.txt changes
NA
## Other PRs 
yes. store related PR

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala [hunk 1]
```diff
@@ -169,7 +169,8 @@ class BaseColumnFormatRelation(
   override def cachedBatchAggregate(batch: CachedBatch): Unit = {
     // if number of rows are greater than columnBatchSize then store
     // otherwise store locally
-    if (batch.numRows >= Constant.COLUMN_MIN_BATCH_SIZE) {
+    if (batch.numRows >= Constant.COLUMN_MIN_BATCH_SIZE ||
+        java.lang.Boolean.getBoolean("forceFlush")) {
```
Comment (kneeraj) [inline core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala]: Is this a new property you have added? How is this related to this change?

Comment (rishitesh) [inline core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala]: This is test flag. We need to test some cases where we intend to flush always.

File: core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala [hunk 2]
```diff
@@ -58,6 +59,28 @@ final class JDBCSourceAsColumnarStore(_connProperties: ConnectionProperties,
...
+      batchId: UUID, partitionId: Int): (Connection => Any) = {
+    {
+      (connection: Connection) => {
+        super.doInsert(tableName, batch, batchId, partitionId)(connection)
+        connectionType match {
+          case ConnectionType.Embedded =>
+            val resolvedName = ExternalStoreUtils.lookupName(tableName,
+              connection.getSchema)
+            val region = Misc.getRegionForTable(resolvedName, true)
+            region.asInstanceOf[AbstractRegion] match {
```
Comment (kneeraj) [inline core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala]: What is the plan for Split for later? Can you file a jira just for a place holder. Later we can update it with the approach.

File: core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala (hunk 1 above)
Comment (kneeraj) [inline core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala]: Just for testing?

Comment (rishitesh) [inline core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala]: Yes. Our scala tests uses a batch size of 3. if we use this flag it will always insert in cached batch table. Later we can validate number of rows which landed in column table.

File: core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala (hunk 2 above)
Comment (rishitesh) [inline core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala]: Yes sure. Will add a JIRA.

**Comment by kneeraj** [comment]
Looks good.

File: core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala (hunk 2 above)
Comment (rishitesh) [inline core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala]: https://jira.snappydata.io/browse/SNAP-1048 added.
